.changelog/3333.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
+resource/mongodbatlas_advanced_cluster: Adds `delete_on_create_timeout` for deleting cluster in create when timeout hits
+```
+
+```release-note:enhancement
+resource/mongodbatlas_advanced_cluster (preview provider 2.0.0): Adds `delete_on_create_timeout` for deleting cluster in create when timeout hits
```

Comment:
Adds `delete_on_create_timeout` for deleting a cluster when the cluster creation fails and timeout is reached.